### ISSUE TITLE
Add multiple loss implementations

### ISSUE THREAD

**Issue by yawenzzzz** [allenai/olmoearth_pretrain#12]
Include multiple loss functions for pretraining and evaluation, such as CrossEntropy, MAE/MSE, and PatchDiscrimination.

References: 
Presto - https://github.com/nasaharvest/presto-v3/blob/main/src/loss.py
Anysat - https://github.com/gastruc/AnySat/blob/main/src/models/losses.py

Ideally, we would like to have a losses meta-object that can take a mix of losses, which improve flexibility in terms of pretraining objective.
